### PR TITLE
feat: blog2/main

### PR DESCRIPTION
**Pull request by jacoobes**
adds the blog for 2.0 release and updates typedoc

### CODE REVIEW

**Comment by SrIzan10** [comment]
LOOK MOM IM FAMOUS 😎

**Review by SrIzan10 — CHANGES_REQUESTED** [review]
```
<img src="https://img.srizan.ml/Discord_z8Sn1UBfEe.png" />
All deprecation warnings from previous versions have taken effect, and are removed in 2.0.
```
should be
```
<img src="https://img.srizan.ml/Discord_z8Sn1UBfEe.png" />

All deprecation warnings from previous versions have taken effect, and are removed in 2.0.
```

**Review by SrIzan10 — APPROVED** [review]

File: blog/2022-12-13-mdx-blog-post.md [hunk 1]
```diff
@@ -41,5 +42,150 @@ export default eventModule({
 ```
 ## Dependency Injection and Decoupling
 
-THIS SECTION IS PROBABLY THE MOST IMPORTANT!! <br/>
-i should have eaten something before i feel hungry
```
Comment (Murtatrxx) [inline blog/2022-12-13-mdx-blog-post.md:45]: 🤨

**Review by Murtatrxx — APPROVED** [review]

**Comment by Murtatrxx** [comment]
I'll fix all small things later, but it can be merged.

**Pull request merged**